src/structures/CategoryChannel.js

```diff
@@ -26,6 +26,35 @@ class CategoryChannel extends GuildChannel {
    * @param {SetParentOptions} [options={}] The options for setting the parent
    * @returns {Promise<GuildChannel>}
    */
+
+  /**
+   * Options for creating a channel using {@link CategoryChannel#createChannel}.
+   * @typedef {Object} CategoryCreateChannelOptions
+   * @property {ChannelType|number} [type='GUILD_TEXT'] The type of the new channel.
+   * @property {string} [topic] The topic for the new channel
+   * @property {boolean} [nsfw] Whether the new channel is nsfw
+   * @property {number} [bitrate] Bitrate of the new channel in bits (only voice)
+   * @property {number} [userLimit] Maximum amount of users allowed in the new channel (only voice)
+   * @property {OverwriteResolvable[]|Collection<Snowflake, OverwriteResolvable>} [permissionOverwrites]
+   * Permission overwrites of the new channel
+   * @property {number} [position] Position of the new channel
+   * @property {number} [rateLimitPerUser] The ratelimit per user for the new channel
+   * @property {string} [reason] Reason for creating the new channel
+   */
+
+  /**
+   * Creates a new channel on this category.
+   * <info>You cannot create a channel of type `GUILD_CATEGORY` inside a CategoryChannel.</info>
+   * @param {string} name The name of the new channel
+   * @param {CategoryCreateChannelOptions} options Options for creating the new channel
+   * @returns {Promise<GuildChannel>}
+   */
+  createChannel(name, options = {}) {
```

Comment:
Maybe don't default options

And before you say, yes, it'll be fine
![image](https://user-images.githubusercontent.com/17960496/132722902-433f3ac7-540c-4429-adb3-11decb8d9f67.png)

Reply:
i think there are a few places where
`{ ...(x ?? {}) }` is used perhaps that could be changed if undefined is valid

Reply:
Never seen that notation Sugden but I’ll change it since the main method already defaults these to an empty object